bor/README.md

```diff
@@ -0,0 +1,166 @@
+Finally, if there are 2/3+ `YES` votes, the `PostHandleMsgEventSpan` persists the proposed span in the state via the keeper :  
+
+```
+err := k.FreezeSet(ctx, msg.ID, msg.StartBlock, msg.EndBlock, msg.ChainID, msg.Seed)
+if err != nil {
+	k.Logger(ctx).Error("Unable to freeze validator set for span", "Error", err)
+	return common.ErrUnableToFreezeValSet(k.Codespace()).Result()
+}
+```
+
+`FreezeSet` internally invokes `SelectNextProducers`, which pseudo-randomly picks producers from the validaor set, leaning more towards validators with higher stake:
+
+```
+// select next producers
+newProducers, err := k.SelectNextProducers(ctx, seed)
+if err != nil {
+	return err
+}
+```
+
+and then intialises and stores the span:
+
+```
+newSpan := hmTypes.NewSpan(
+	id,
+	startBlock,
+	endBlock,
+	k.sk.GetValidatorSet(ctx),
+	newProducers,
+	borChainID,
+)
+
+return k.AddNewSpan(ctx, newSpan)
+```
+
+## HOW TO PROPOSE A SPAN ?
+
+A validator can leverage the CLI to propose a span like so :
+
+```
+heimdallcli tx bor propose-span --proposer <VALIDATOR ADDRESS> --start-block <START_BLOCK> --span-id <SPAN_ID> --bor-chain-id <BOR_CHAIN_ID>
+```
+
+Or the rest server : 
+
+```
+curl -X POST "localhost:1317/bor/propose-span?bor-chain-id=<BOR_CHAIN_ID>&start-block=<START_BLOCK>&span-id=<SPAN_ID>"
+```
+
+## QUERY COMMANDS
+
+One can run the following query commands from the bor module :
+
+* `span` - Query the span corresponding to the given span id:
+
+via CLI
+```
+heimdallcli query bor span --span-id=<span ID here>
+```
+
+via Rest
+```
+curl localhost:1317/bor/span/1000
+```
+
+* `latest span` - Query the latest span : 
+
+via CLI
+```
+heimdallcli query bor latest-span
+```
+
+via Rest
+```
+curl localhost:1317/bor/latest-span
+```
+
+* `params` - Fetch the parameters associated to bor module :
+
+via CLI
+```
+heimdallcli query bor params
+```
+
+via Rest
+```
+curl localhost:1317/bor/params
+```
+
+* `spanlist` - Fetch span list :
+
+via CLI
+```
+heimdallcli query bor spanlist --page=<page number here> --limit=<limit here>
+```
+
+* `next-span-seed` - Query the seed for the next span :
+
+via CLI
+```
+heimdallcli query bor next-span-seed
+```
+
+via Rest 
+```
+curl localhost:1317/bor/next-span-seed
+```
+
+* `propose-span` - Print the `propose-span` command :
+
+via CLI
+```
+heimdallcli query bor propose-span --proposer <VALIDATOR ADDRESS> --start-block <START_BLOCK> --span-id <SPAN_ID> --bor-chain-id <BOR_CHAIN_ID>
+```
+
+via Rest
+```
+curl "localhost:1317/bor/prepare-next-span?span_id=23&start_block=1000&chain_id="137""
```

Comment:
Start block and chain id are hardcoded here. Is it intentional?

Reply:
Maybe. I was probably just caught in two minds whether to add an example or a generalized query 😄 
I'll rectify it, thanks!